libraries/Network/src/NetworkManager.cpp

```diff
@@ -97,32 +61,44 @@ int NetworkManager::hostByName(const char* aHostname, IPAddress& aResult, bool p
     }
 
     aResult = static_cast<uint32_t>(0);
-    params.hostname = aHostname;
-    params.addr_type = (preferV6 || hasGlobalV6)?LWIP_DNS_ADDRTYPE_IPV6_IPV4:LWIP_DNS_ADDRTYPE_IPV4;
-    params.result = 0;
-    aResult.to_ip_addr_t(&(params.addr));
 
+    // First check if the host parses as a literal address
     if (!aResult.fromString(aHostname)) {
-        Network.waitStatusBits(NET_DNS_IDLE_BIT, 16000);
-        Network.clearStatusBits(NET_DNS_IDLE_BIT | NET_DNS_DONE_BIT);
```

Comment:
Removing any reference to `NET_DNS_IDLE_BIT` and `NET_DNS_DONE_BIT` is also necessary. And if `Network.[get|set|wait]StatusBits` are only used for those two flags, then we can probably get rid of that too.